File: frontend/apps/web/lib/services/email.ts
import { apiClient } from './api-client'

export const EMAIL_PROVIDERS = [
    { id: 'microsoft', name: 'Microsoft 365', logo: 'microsoft', icon: 'M' },
    { id: 'gmail', name: 'Google Workspace', logo: 'google', icon: 'G' },
    { id: 'zoho', name: 'Zoho Mail', logo: 'zoho', icon: 'Z' },
    { id: 'smtp', name: 'Custom SMTP', logo: 'smtp', icon: '@' },
    { id: 'sendgrid', name: 'SendGrid', logo: 'sendgrid', icon: 'S' },
    { id: 'ses', name: 'AWS SES', logo: 'aws', icon: 'A' }
]

export interface EmailSettings {
    id: string
    provider: string
    smtp_host: string
    smtp_port: number
    smtp_use_tls: boolean
    smtp_use_ssl: boolean
    smtp_username: string
    smtp_password?: string // Only for creating/updating
    from_email: string
    from_name: string
    reply_to_email?: string
    is_active: boolean
    test_email?: string
    last_test_sent_at?: string
    last_test_status?: 'success' | 'failed'
    last_test_error?: string
    created_at: string
    updated_at: string
}

export interface EmailTemplate {
    id: string
    template_key: string
    subject: string
    body_html: string
    body_plain: string
    variables?: string[]
    is_active: boolean
    created_at: string
    updated_at: string
}

export interface TestEmailRequest {
    target_email: string
}

export interface TestEmailResponse {
    success: boolean
    message: string
}

export const emailService = {
    /**
     * Get all email settings (configurations)
     */
    async getSettings(): Promise<EmailSettings[]> {
        const response = await apiClient.get('/admin/email/email-settings')
        return Array.isArray(response) ? response : []
    },

    /**
     * Get active email configuration
     */
    async getActiveSettings(): Promise<EmailSettings | null> {
        try {
            return await apiClient.get('/admin/email/email-settings/active')
        } catch (error) {
            return null
        }
    },

    /**
     * Create new email settings
     */
    async createSettings(data: Partial<EmailSettings>): Promise<EmailSettings> {
        return await apiClient.post('/admin/email/email-settings', data)
    },

    /**
     * Update email settings
     */
    async updateSettings(id: string, data: Partial<EmailSettings>): Promise<EmailSettings> {
        return await apiClient.put(`/admin/email/email-settings/${id}`, data)
    },

    /**
     * Delete email settings
     */
    async deleteSettings(id: string): Promise<void> {
        await apiClient.delete(`/admin/email/email-settings/${id}`)
    },

    /**
     * Test email configuration
     */
    async testSettings(id: string, email: string): Promise<TestEmailResponse> {
        return await apiClient.post(`/admin/email/email-settings/${id}/test`, { test_email: email })
    },

    /**
     * Activate specific email configuration
     */
    async activateSettings(id: string): Promise<EmailSettings> {
        return await apiClient.post(`/admin/email/email-settings/${id}/activate`)
    },

    /**
     * Get all email templates
     */
    async getTemplates(): Promise<EmailTemplate[]> {
        const response = await apiClient.get('/admin/email/email-templates')
        return Array.isArray(response) ? response : []
    }
}

File: backend/app/models/email.py
"""Email system models"""
from sqlalchemy import Column, String, Integer, Boolean, Text, DateTime, ForeignKey
from sqlalchemy.dialects.postgresql import UUID, JSONB
from sqlalchemy.sql import func
from app.models.base import Base
import uuid


class EmailSettings(Base):
    """Email/SMTP configuration settings"""
    __tablename__ = "email_settings"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    provider = Column(String(50), nullable=False, comment="microsoft, gmail, zoho, smtp")
    
    # SMTP Configuration
    smtp_host = Column(String(255), nullable=False)
    smtp_port = Column(Integer, nullable=False, default=587)
    smtp_use_tls = Column(Boolean, nullable=False, default=True)
    smtp_use_ssl = Column(Boolean, nullable=False, default=False)
    
    # Authentication
    smtp_username = Column(String(255), nullable=False)
    smtp_password_encrypted = Column(Text, nullable=False, comment="Encrypted SMTP password")
    from_email = Column(String(255), nullable=False)
    from_name = Column(String(255), nullable=False, default="JCTC System")
    reply_to_email = Column(String(255), nullable=True)
    
    # Testing & Status
    is_active = Column(Boolean, nullable=False, default=False)
    test_email = Column(String(255), nullable=True)
    last_test_sent_at = Column(DateTime(timezone=True), nullable=True)
    last_test_status = Column(String(50), nullable=True, comment="success, failed")
    last_test_error = Column(Text, nullable=True)
    
    # Metadata
    created_at = Column(DateTime(timezone=True), nullable=False, server_default=func.now())
    updated_at = Column(DateTime(timezone=True), nullable=False, server_default=func.now(), onupdate=func.now())
    created_by = Column(UUID(as_uuid=True), ForeignKey('users.id'), nullable=True)


class EmailTemplate(Base):
    """Email templates for system notifications"""
    __tablename__ = "email_templates"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    template_key = Column(String(100), unique=True, nullable=False, comment="user_invite, password_reset, etc.")
    subject = Column(String(255), nullable=False)
    body_html = Column(Text, nullable=False)
    body_plain = Column(Text, nullable=True)
    variables = Column(JSONB, nullable=True, comment="List of allowed template variables")
    is_active = Column(Boolean, nullable=False, default=True)
    created_at = Column(DateTime(timezone=True), nullable=False, server_default=func.now())
    updated_at = Column(DateTime(timezone=True), nullable=False, server_default=func.now(), onupdate=func.now())
